### PR TITLE
ignore wait4 errs in observe max-err-count

### PR DESCRIPTION
**Pull request by juanvallejo**
Fixes https://github.com/openshift/origin/issues/17743

These errors should not count against the --maximum-errors count
as the process has cleanly run and exited before wait4 syscall is made.

cc @smarterclayton @soltysh

### CODE REVIEW

File: pkg/oc/cli/cmd/observe/observe.go [hunk 1]
```diff
@@ -716,6 +718,13 @@ func measureCommandDuration(m *prometheus.SummaryVec, fn func() error, labels ..
 		statusCode = -1
 	}
 	m.WithLabelValues(append(labels, strconv.Itoa(statusCode))...).Observe(float64(duration / time.Millisecond))
+
+	if err != nil && err.Error() == errNoChildProc {
```
Comment (soltysh) [inline pkg/oc/cli/cmd/observe/observe.go]: I'd rather not hardcode the value like that. See if you can get some reasonable information from the error, when being cast to [exec.ExitError](https://golang.org/pkg/os/exec/#ExitError). I think you should be able to get `ProcessState.Sys` which 'returns system-dependent exit information about the process. Convert it to the appropriate underlying type, such as syscall.WaitStatus on Unix, to access its contents. ' from there you should be able to get some system-wide constant.

Comment (liggitt) [reply pkg/oc/cli/cmd/observe/observe.go]: agree, this isn't a reasonable fix

File: pkg/oc/cli/cmd/observe/observe.go [hunk 2]
```diff
@@ -716,9 +716,28 @@ func measureCommandDuration(m *prometheus.SummaryVec, fn func() error, labels ..
 		statusCode = -1
 	}
 	m.WithLabelValues(append(labels, strconv.Itoa(statusCode))...).Observe(float64(duration / time.Millisecond))
+
+	errno := errnoError(err)
+	if errno == syscall.ECHILD {
```
Comment (soltysh) [inline pkg/oc/cli/cmd/observe/observe.go]: Since you've already created that method just:
```go
if errnoError(err) == syscall.ECHILD {
...
```

**Comment by juanvallejo** [comment]
@soltysh thanks, comment addressed

File: pkg/oc/cli/cmd/observe/observe.go [hunk 3]
```diff
@@ -716,9 +716,31 @@ func measureCommandDuration(m *prometheus.SummaryVec, fn func() error, labels ..
...
+		// that the subprocess has started and ended
+		// before the wait call was made.
+		return nil
+	}
+
 	return err
 }
 
+func errnoError(err error) syscall.Errno {
+	if err == nil {
```
Comment (soltysh) [inline pkg/oc/cli/cmd/observe/observe.go]: How about:

```go
if se, ok := err.(*os.SyscallError); ok {
    if errno, ok := se.Err.(syscall.Errno); ok {
        return errno
    }
}

return 0
```

Comment (juanvallejo) [reply pkg/oc/cli/cmd/observe/observe.go]: thanks, done

**Pull request merged**